### PR TITLE
VC DI proof request

### PR DESCRIPTION
**Pull request by sarthakvijayvergiya**
Presentation request (DIF) for VC_DI credentials

### CODE REVIEW

**Comment by ianco** [comment]
I know this is still WIP, but we need to have a description on the PR (is necessary in case anyone needs to look back on this PR in the future, and is also used for release notes).  Also it's good - if the PR is in WIP state - to make a note of what is done and what's still outstanding ...

File: aries_cloudagent/protocols/present_proof/v2_0/routes.py [hunk 1]
```diff
@@ -572,6 +572,8 @@ async def present_proof_credentials_list(request: web.BaseRequest):
                     extra_query,
                 )
             )
+        indy_credentials = await indy_holder.get_credentials(start, count, {})
```
[the remark this answers is not in the document]
Comment (ianco) [reply aries_cloudagent/protocols/present_proof/v2_0/routes.py:575]: Not sure what this extra line is for, but it's breaking the "indy" credential scenario

**Comment by ianco** [comment]
Also just noticed the commit is missing the DCO signoff

File: demo/runners/agent_container.py [hunk 2]
```diff
@@ -453,6 +455,8 @@ async def handle_present_proof_v2_0(self, message):
                 if not pres_request_dif and not pres_request_indy:
                     raise Exception("Invalid presentation request received")
 
+                # TODO for a DIF presentation need to check if it is asking for an anoncreds proof ...
```
Comment (ianco) [inline demo/runners/agent_container.py:458]: This comment can be removed

File: aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py [hunk 3]
```diff
@@ -169,7 +171,7 @@ async def create_pres(
             domain = proof_request["options"].get("domain")
         if not challenge:
             challenge = str(uuid4())
-
+        # TODO handle vc_di format in the future
```
Comment (ianco) [inline aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py:174]: This comment can be removed

**Review by ianco — CHANGES_REQUESTED** [review]
A couple of comments to be removed (minor issue) and unit test coverage (SonarCloud reports 27 new lines in each of `aries_cloudagent/anoncreds/holder.py` and `aries_cloudagent/vc/vc_di/prove.py` that are not covered - I think 2 simple "happy path" tests could cover most of this).

Also PR description ...  (@EmadAnwer  not sure if you mentioned you can't edit the description?)

Overall looks really good!

**Review by ianco — APPROVED** [review]

**Comment by jamshale** [comment]
There is quite a few things reported in sonarcloud. I don't think everything needs to be addressed but some definitely should like the unused variable and shadow variable names.

https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true

**Comment by ianco** [comment]
> There is quite a few things reported in sonarcloud. I don't think everything needs to be addressed but some definitely should like the unused variable and shadow variable names.
> 
> https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true
> 
> Maybe have a go over and see if we can make some improvements. I think overall the code looks good and well tested.

Good catch @jamshale 

@EmadAnwer  can you review the SonarCloud report and do a bit of cleanup?  As Jamie says we don't need to address everything but take a pass and see what you think.

**Review by ianco — CHANGES_REQUESTED** [review]
SonarCloud as noted by Jamie

**Comment by EmadAnwer** [comment]
> > There is quite a few things reported in sonarcloud. I don't think everything needs to be addressed but some definitely should like the unused variable and shadow variable names.
> > https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true
> > Maybe have a go over and see if we can make some improvements. I think overall the code looks good and well tested.
> 
> Good catch @jamshale
> 
> @EmadAnwer can you review the SonarCloud report and do a bit of cleanup? As Jamie says we don't need to address everything but take a pass and see what you think.

yes sure

**Comment by EmadAnwer** [comment]
> ## [Quality Gate Passed](https://sonarcloud.io/dashboard?id=hyperledger_aries-cloudagent-python&pullRequest=2960) **Quality Gate passed**
> Issues  [12 New issues](https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true)  [0 Accepted issues](https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolutions=WONTFIX)
> 
> Measures  [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true)  [86.4% Coverage on New Code](https://sonarcloud.io/component_measures?id=hyperledger_aries-cloudagent-python&pullRequest=2960&metric=new_coverage&view=list)  [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=hyperledger_aries-cloudagent-python&pullRequest=2960&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=hyperledger_aries-cloudagent-python&pullRequest=2960)



> > > There is quite a few things reported in sonarcloud. I don't think everything needs to be addressed but some definitely should like the unused variable and shadow variable names.
> > > https://sonarcloud.io/project/issues?id=hyperledger_aries-cloudagent-python&pullRequest=2960&resolved=false&sinceLeakPeriod=true
> > > Maybe have a go over and see if we can make some improvements. I think overall the code looks good and well tested.
> > 
> > 
> > Good catch @jamshale
> > @EmadAnwer can you review the SonarCloud report and do a bit of cleanup? As Jamie says we don't need to address everything but take a pass and see what you think.
> 
> yes sure

@jamshale @ianco I have fixed most of it, check it please, let me know if there are any suggestions

**Review by ianco — APPROVED** [review]

**Review by jamshale — APPROVED** [review]

**Comment by jamshale** [comment]
I'm going to merge this. Would be nice to have in the 1.0.0 release and don't predict any regressions.

**Comment by swcurran** [comment]
w00t!!!  Awesome.

**Pull request merged**